backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/migration/MigrationTaskDTO.java

```diff
@@ -0,0 +1,40 @@
+package org.apache.james.backends.cassandra.migration;
+
+import java.util.function.Function;
+
+import org.apache.james.backends.cassandra.versions.SchemaVersion;
+import org.apache.james.json.DTOModule;
+import org.apache.james.server.task.json.dto.TaskDTO;
+import org.apache.james.server.task.json.dto.TaskDTOModule;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.annotations.VisibleForTesting;
+
+class MigrationTaskDTO implements TaskDTO {
+
+    @VisibleForTesting static final Function<MigrationTask.Factory, TaskDTOModule<MigrationTask, MigrationTaskDTO>> MODULE =
+            factory -> DTOModule.forDomainObject(MigrationTask.class)
+                .convertToDTO(MigrationTaskDTO.class)
+                .toDomainObjectConverter(dto -> factory.create(new SchemaVersion(dto.targetVersion)))
+                .toDTOConverter((task, type) -> new MigrationTaskDTO(type, task.getTargetVersion().getValue()))
```

Comment:
You exposed the targetVersion in the Task. You should create the DTO as an inner class of the task.
this way you will not have to expose it.

Reply:
changed